modules/consentManagementUsp.js

```diff
@@ -0,0 +1,309 @@
+  }
+
+  if (!includes(Object.keys(uspCallMap), userUSP)) {
+    utils.logWarn(`USP framework (${userUSP}) is not a supported framework. Aborting consentManagement module and resuming auction.`);
+    return hookConfig.nextFn.apply(hookConfig.context, hookConfig.args);
+  }
+
+  uspCallMap[userUSP].call(this, processUspData, uspapiFailed, hookConfig);
+
+  // only let this code run if module is still active (ie if the callbacks used by USPs haven't already finished)
+  if (!hookConfig.haveExited) {
+    if (consentTimeout === 0) {
+      processUspData(undefined, hookConfig);
+    } else {
+      hookConfig.timer = setTimeout(uspapiTimeout.bind(null, hookConfig), consentTimeout);
+    }
+  }
+}
+
+/**
+ * This function checks the consent data provided by USPAPI to ensure it's in an expected state.
+ * If it's bad, we exit the module depending on config settings.
+ * If it's good, then we store the value and exits the module.
+ * @param {object} consentObject required; object returned by USPAPI that contains user's consent choices
+ * @param {object} hookConfig contains module related variables (see comment in requestBidsHook function)
+ */
+function processUspData(consentObject, hookConfig) {
+  const valid = !!(consentObject && consentObject.usPrivacy);
+  if (!valid) {
+    uspapiFailed(`UPSAPI returned unexpected value during lookup process.`, hookConfig, consentObject);
+    return;
+  }
+
+  clearTimeout(hookConfig.timer);
+  storeUspConsentData(consentObject);
+  exitModule(null, hookConfig);
+}
+
+/**
+ * General timeout callback when interacting with USPAPI takes too long.
+ */
+function uspapiTimeout(hookConfig) {
+  cmpFailed('USPAPI workflow exceeded timeout threshold.', hookConfig);
+}
+
+/**
+ * This function contains the controlled steps to perform when there's a problem with USPAPI.
+ * @param {string} errMsg required; should be a short descriptive message for why the failure/issue happened.
+ * @param {object} hookConfig contains module related variables (see comment in requestBidsHook function)
+ * @param {object} extraArgs contains additional data that's passed along in the error/warning messages for easier debugging
+*/
+function uspapiFailed(errMsg, hookConfig, extraArgs) {
+  clearTimeout(hookConfig.timer);
+
+  // still set the consentData to undefined when there is a problem as per config options
+  if (allowAuction) {
+    storeUspConsentData(undefined);
+  }
+
+  exitModule(errMsg, hookConfig, extraArgs);
+}
+
+/**
+ * Stores USP data locally in module and then invokes uspDataHandler.setConsentData() to make information available in adaptermanger.js for later in the auction
+ * @param {object} cmpConsentObject required; an object representing user's consent choices (can be undefined in certain use-cases for this function only)
+ */
+function storeUspConsentData(consentObject) {
+  consentData = { usPrivacy: consentObject ? consentObject.usPrivacy : undefined };
+  uspDataHandler.setConsentData(consentData);
```

Comment:
I don't think this first line works (`usPrivacy: consentObject`).  The entire thing should probably just be changed to
```
if (consentObject && consentObject.usPrivacy) {
   uspDataHandler.setConsentData(consentObject.usPrivacy);
}
```
no need to set it if it doesn't exist